### DISCUSSION TITLE
Should i commit the `gen` folder?

### DISCUSSION THREAD

**Discussion by nooblyf** [tauri-apps/tauri#8323] [Q&A] [answered]
When building apps for mobile after doing `pnpm tauri ios init` a folder named `gen` is created with bunch of stuff, is it best practice to commit it to the github repo?

**Comment by FabianLars** [comment, marked as answer]
Generally yes. it has .gitignore folders inside that ignores the build artifacts etc. The gen folder also contains the normal android studio / xcode projects which you can modify if you want which would require you to commit it.

**Reply by mcitem** [reply]
Is there a way to customize the output directory for the 'gen' folder, similar to how the dist directory is configured in Vite? I couldn't find any relevant configuration options in the documentation.

**Reply by FabianLars** [reply]
Nope

**Reply by LukasWillin** [reply]
I hope I am not unnecessarily bringing this discussion back to live. I think having the directory being named "gen" is confusing to many, since (or at least I) was told again and again during my studies (and later) not to commit generated output. :P
Just food for thought.

**Comment by pronebird** [comment]
@FabianLars Would you mind elaborating on that?

Here is the list of folders that I found being generally speaking a development junk:

- `gen/schemas` - Lotta json that aren't even pretty printed to be considered human readable. Can we add those in .gitignore given that we don't develop for mobile, i.e no Xcode files in there etc?
- In plugins:`permissions/{autogenerated, schemas}` - very much junk too, doesn't add any value, some toml files with plugin commands generated etc..  It could be used somewhere but it as well may be regenerated if needed. Is that correct?

**Reply by FabianLars** [reply]
> gen/schemas - Lotta json that aren't even pretty printed to be considered human readable. Can we add those in .gitignore given that we don't develop for mobile, i.e no Xcode files in there etc?

Yes, and we have, or will update create-tauri-app to include it in .gitignore by default like tauri init does. That said, comitting it is fine and iirc we will change it so that all files are prettified (currently some are, some are not).

> In plugins:permissions/{autogenerated, schemas} - very much junk too, doesn't add any value, some toml files with plugin commands generated etc.. It could be used somewhere but it as well may be regenerated if needed. Is that correct?

These should be committed (at least if you plan on publishing your plugin).

**Comment by pronebird** [comment]
> > gen/schemas - Lotta json that aren't even pretty printed to be considered human readable. Can we add those in .gitignore given that we don't develop for mobile, i.e no Xcode files in there etc?
> 
> Yes, and we have, or will update create-tauri-app to include it in .gitignore by default like tauri init does. That said, comitting it is fine and iirc we will change it so that all files are prettified (currently some are, some are not).
> 
> > In plugins:permissions/{autogenerated, schemas} - very much junk too, doesn't add any value, some toml files with plugin commands generated etc.. It could be used somewhere but it as well may be regenerated if needed. Is that correct?
> 
> These should be committed (at least if you plan on publishing your plugin).

Alright, thanks! So we actually don't plan on publishing our plugins as we have them deployed privately/in-house as a part of our workspace, so we seemingly successfully removed and ignored the following files:

```
# Tauri app ignore
src-tauri/gen

# Per plugin ignore
plugin/permissions/schemas
plugin/permissions/autogenerated
```
 
So far so good. 🚀
